packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -1104,6 +1137,48 @@ export class FluidDataStoreRuntime
 			throw new LoggingError("Runtime is closed");
 		}
 	}
+
+	/**
+	 * Summarizer client should not have local changes. These changes can become part of the summary and can break
+	 * eventual consistency. For example, the next summary (say at ref seq# 100) may contain these changes whereas
+	 * other clients that are up-to-date till seq# 100 may not have them yet.
+	 */
+	private identifyLocalChangeInSummarizer(
+		eventName: string,
+		channelId: string,
+		channelType: string,
+	) {
+		if (this.clientDetails.type === "summarizer") {
+			// If the count of telemetry logged has crossed the threshold, don't log any more.
+			if (
+				this.localChangesTelemetryThreshold !== undefined &&
+				this.localChangesTelemetryCount >= this.localChangesTelemetryThreshold
+			) {
+				return;
+			}
+			this.localChangesTelemetryCount++;
+
+			// Log a telemetry if there are local changes in the summarizer. This will give us data on how often
+			// this is happening and which data stores do this. The eventual goal is to disallow local changes
+			// in the summarizer and the data will help us plan this.
+			this.mc.logger.sendTelemetryEvent({
+				eventName,
+				channelType,
+				channelId: {
+					value: channelId,
+					tag: TelemetryDataTag.CodeArtifact,
+				},
+				fluidDataStoreId: {
+					value: this.id,
+					tag: TelemetryDataTag.CodeArtifact,
+				},
+				fluidDataStorePackagePath: packagePathToTelemetryProperty(
+					this.dataStoreContext.packagePath,
+				),
+				stack: generateStack(),
```

Comment:
I think it's a plan, but worth calling out - we should reduce in future amount of code associated with checks with small set of cases where we throw UserError objects on key paths.